### ISSUE TITLE
Make font-locking of json work when using jq-interactively

### ISSUE THREAD

**Issue by ljos** [ljos/jq-mode#7]
See code from [305ead8#L168](https://github.com/ljos/jq-mode/blob/305ead8f087b1b817291d5f0365e5226199ec58c/jq-mode.el#L168) and onwards.

We are trying to use the same buffer as the json file that we are editing, but we don't want to change the content of the file before the user commits. We use an overlay to hide the original text and show the edited text. The problem is that we don't get font-locking for the edited text. It would be nice if this was possible.

**Comment by mcraveiro** [comment]
One possible solution - at least for interactive work - would be to use an "output" buffer, like the deprecated jq.el?[1] I find that I always end up undoing my JSON buffer to get back to where I was after using ```jq-interactively```.

Cheers

Marco

[1] https://github.com/rejeep/jq.el

**Comment by ljos** [comment]
`jq-interactively` started as a partial port and reimplementation of jq.el. 

`jq-interactively` doesn't actually change the buffer before you commit the command. It is using an overlay to hide the original text and show a different text. However, the different text doesn't get font-locked.

If you press `C-g` while in the minibuffer with `jq-interactively` it should revert the buffer to the previous stage. If you press `RET` it will commit the change.

Having a output-buffer is a different approach and one that is possible to implement. I just didn't like that way when I started, but maybe it is less confusing for others to use. I will have to think about it.

I could also support both, but I don't know when I will make the time; that was why I added the issue so if I am lucky someone else will want to look at it.
